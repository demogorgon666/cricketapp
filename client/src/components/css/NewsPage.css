@import url("https://fonts.googleapis.com/css?family=Montserrat:400,500,600&display=swap");

* {
  font-family: Montserrat, sans-serif;
}
.newspage-grid-container {
  display: grid;
  grid-template-columns: 2fr 1fr;
  background-color: white;
  /* padding: 10px; */
  color: black;
  padding: 100px;
  background-color: #ecf0f1;
  /* font-size: 30px; */

  /* padding-right: 100px;
  padd */
}
#recent-news-title {
  font-size: 15px;
}
#news-date {
  margin-left: 25px;
}
.div-newspage {
  width: 100%;
  height: 140px;
  /* border: 1px solid #707070; */
  border-radius: 8px;
  margin-bottom: 30px;
  display: flex;
  background-color: white;
}
.newspage-grid-container p {
  color: black;
  display: grid;
  /* padding: 10px; */
}
.grid-item {
  color: black;
  background-color: #ecf0f1;
  /* margin-top: 50px; */
  /* border: 1px solid black; */
  /* margin-left: 180px; */
  column-gap: 400px;
}
.grid-item2 {
  color: black;
  background-color: white;
  /* margin-top: 50px; */
  /* border: 1px solid black; */
  column-gap: 400px;
  padding: 15px;
  /* margin-left: 10px; */
  text-align: left;
  margin: 20px 60px 10px 10px;
}
#newspage-news-title {
  /* margin-top: 70px; */
  font-size: 35px;
}

.main_img {
  width: 100%;
  border-radius: 8px;
}

.recent-news {
  width: 100%;
  border: 1px solid #707070;
  border-radius: 8px;
  margin-bottom: 30px;
  display: flex;
}
.date {
  font-size: 12px;
}

@media screen and (max-width: 480px) {
  .newspage-grid-container {
    display: grid;
    grid-template-columns: 2fr;
    /* background-color: white; */
    /* padding: 10px; */
    /* color: black; */
    padding: 0px;

    /* padding: 100px; */

    /* padding-right: 100px;
        padd */
  }
  /* .div-newspage {
    border: 1px solid #707070;
  } */

  #newspage-news-title {
    margin: 50px 0px 0px 0px;
    font-size: 30px;
    border: 0px;
  }
  #news-date {
    margin: 10px 0px 10px 0px;
  }
  .main_img {
    width: 100%;
  }
  .grid-item {
    color: black;
    background-color: #ecf0f1;
    /* margin-top: 50px; */
    /* border: 1px solid black; */
    /* margin-left: 180px; */
    /* column-gap: 400px; */
  }
  .grid-item2 {
    color: black;
    background-color: white;
    /* border: 1px solid black; */
    padding: 5px;
    margin: 0px 0px 0px 0px;
  }
  .newspage-grid-container p {
    padding: 0px 10px 0px 10px;
  }
}
@media screen and (max-width: 768px) {
  .newspage-grid-container {
    display: grid;
    grid-template-columns: 2fr;
    padding: 0px;
  }
  .grid-item {
    padding: 10px;
  }
}
