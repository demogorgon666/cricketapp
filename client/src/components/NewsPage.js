import React, { Component } from "react";
import { getNewsById } from "../actions/NewsAction";

import { getNews } from "../actions/Home";
import { connect } from "react-redux";
import "./css/NewsPage.css";
// import Navbar from './common/Navbar'
import news_img from "../components/images/vk.jpeg";
import { Link } from "react-router-dom";
import "./css/Home.css";

export class NewsPage extends Component {
  componentDidMount() {
    // console.log('NEWSPAGE')
    // console.log(this.props.match.params.id)
    const id = this.props.match.params.id;
    this.props.getNews();

    this.props.getNewsById(id);
  }
  onNewsClick = id => {
    // const id = this.props.match.params.id;
    // console.log('from news click', id)
    this.props.getNewsById(id);
  };
  render() {
    return (
      <div className="parent">
        <div className="navbar">{/* <Navbar /> */}</div>
        <div>
          {this.props.news.map(news => (
            <div className="newspage-grid-container">
              <div className="grid-item2">
                <p className="grid-item2" id="newspage-news-title">
                  <b>{news.news_title}</b>
                </p>
                <p id="news-date">{news.news_date} </p>

                {/* <div
                  className="div-news-images"
                  style={{
                    backgroundImage: `url(${news_img})`,
                  }}
                > */}
                {/* <img className="main_img" src={news_img} /> */}
                <img
                  // id="pImage"
                  // className="playerImage"
                  className="main_img"
                  src={`data:image/jpeg;base64,${news.news_image}`}
                ></img>
                {/* </div> */}
                <p id="news-content">{news.news_content} </p>
              </div>

              {/* Right side recent news section */}

              <div className="grid-item">
                <div>
                  <div
                    className="div-news-images"
                    style={
                      {
                        // backgroundImage: `url(${news_img})`,
                      }
                    }
                  >
                    {/* <img className="news_img" src={news_img} /> */}
                  </div>
                  {/* using the same css from news list in news page */}
                  {/* recent news */}
                  {this.props.home.map(recent => (
                    // <Link
                    //   to={{
                    //     pathname: "/newsbyid/" + recent.news_id,
                    //   }}
                    // >
                    <div className="div-newspage">
                      <div
                        className="div-news-images"
                        style={{
                          backgroundImage: `url(data:image/jpeg;base64,${recent.news_image}`
                        }}
                      />
                      <div
                        className="div-news-details"
                        onClick={() => this.onNewsClick(recent.news_id)}
                      >
                        <p class="p-news">
                          <b id="recent-news-title">{recent.news_title}</b>
                        </p>
                        <p id="recent-news-date" className="date">
                          {recent.news_date}{" "}
                        </p>
                      </div>
                    </div>
                  ))}
                </div>
              </div>
            </div>
          ))}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  news: state.NewsReducer.news,
  home: state.HomeReducer.home
});
export default connect(
  mapStateToProps,
  {
    getNewsById,
    getNews
  }
)(NewsPage);
