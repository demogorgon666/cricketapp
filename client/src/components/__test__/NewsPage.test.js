import React from 'react'
import { shallow, mount } from 'enzyme'
import { NewsPage } from '../NewsPage'

const newspage = jest.fn()
// creating mock fuctions of actions
const getNews = jest.fn()
const getNewsById = jest.fn()

// creating a dummy data that will be used to check the mapping functionality
// news is the reducer which will recieve the actual data but for testing purpose we are passing dummy data and testing wheter that will be mapped properly or not.
const news = [
  {
    news_id: 1,
    news_content: 'news content',
    news_title: 'this is title',
    news_date: '12-12-12'
  }
]
let home = [
  {
    news_id: 1,
    news_content: 'Recent news content',
    news_title: 'Recent news title',
    news_date: '12-12-12'
  }
]

const wrapper = shallow(
  <NewsPage
    // news reducer which contains dummy data
    news={news}
    history={[]}
    // home reducer which contains dummy data
    home={home}
    // for props function
    getNews={getNews}
    getNewsById={getNewsById}
    // passing id in parameters
    match={{ params: { id: 1 } }}
  />
)

describe('Testing of NewsPage Component', () => {
  it('should render the component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  // testing for mapped functions by creating a dummy mapped value above and checking whether the exact text is being mapped or not.
  it('News title should be shown', () => {
    expect(wrapper.find('#newspage-news-title').text()).toBe('this is title')
  })
  it('News content should be present', () => {
    expect(wrapper.find('#news-content').text()).toBe('news content ')
  })
  it('News date should be present', () => {
    expect(wrapper.find('#news-date').text()).toBe('12-12-12 ')
  })
})

describe('Testing of Recent News section', () => {
  // testing for mapped functions by creating a dummy mapped value above and checking whether the exact text is being mapped or not.
  it('Recent news title should be shown', () => {
    expect(wrapper.find('#recent-news-title').text()).toBe('Recent news title')
  })
  it('Date of the recent news should be shown', () => {
    expect(wrapper.find('#recent-news-date').text()).toBe('12-12-12 ')
  })
})
