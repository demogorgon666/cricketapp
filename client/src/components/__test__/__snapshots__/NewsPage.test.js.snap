// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of NewsPage Component should render the component 1`] = `
<div
  className="parent"
>
  <Connect(Navbar)
    changeGender={[Function]}
    showGender={false}
  />
  <div>
    <div
      className="newspage-grid-container"
    >
      <div
        className="grid-item2"
      >
        <p
          id="newspage-news-title"
        >
          <b>
            this is title
          </b>
        </p>
        <p
          id="news-date"
        >
          12-12-12
           
        </p>
        <img
          className="main_img"
          src="data:image/jpeg;base64,undefined"
        />
        <p
          id="news-content"
        >
          news content
           
        </p>
      </div>
      <div
        className="grid-item"
      >
        <div>
          <div
            className="div-news-images"
          />
          <div
            className="div-newspage"
          >
            <div
              className="div-news-images"
              style={
                Object {
                  "backgroundImage": "url(data:image/jpeg;base64,undefined",
                }
              }
            />
            <div
              className="div-news-details"
              onClick={[Function]}
            >
              <p
                class="p-news"
              >
                <b
                  id="recent-news-title"
                >
                  Recent news title
                </b>
              </p>
              <p
                className="date"
                id="recent-news-date"
              >
                12-12-12
                 
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
